Content.Server/GameObjects/EntitySystems/AI/Pathfinding/Pathfinders/AStarPathfindingJob.cs

```diff
@@ -0,0 +1,185 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using Content.Server.GameObjects.EntitySystems.AI.Pathfinding.Queues;
+using Content.Server.GameObjects.EntitySystems.JobQueues;
+using Content.Server.GameObjects.EntitySystems.Pathfinding;
+using Content.Shared.Pathfinding;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+
+namespace Content.Server.GameObjects.EntitySystems.AI.Pathfinding.Pathfinders
+{
+    public class AStarPathfindingJob : Job<Queue<TileRef>>
+    {
+        public static event Action<AStarRouteDebug> DebugRoute;
+
+        private PathfindingNode _startNode;
+        private PathfindingNode _endNode;
+        private PathfindingArgs _pathfindingArgs;
+        private CancellationTokenSource _cancellationToken;
+
+        public AStarPathfindingJob(
+            double maxTime,
+            PathfindingNode startNode,
+            PathfindingNode endNode,
+            PathfindingArgs pathfindingArgs,
+            CancellationTokenSource cancellationToken = null) : base(maxTime)
```

Comment:
You should probably take in a `CancellationToken` here, not a `CancellationTokenSource`.